Automattic/wp-calypso
# Comments: add a timeout for loading

This PR should fix #15891, where we could get infinite placeholders when a site had no comments.

Implementation Notes: Due to how the reducer is currently storing data, I'm having a hard time deriving fetch state without storing more information. To avoid adding `isRequesting` state, I instead opted to show the loader until either we have some comments in the current filter, or we reach our timeout.

Folks are welcome to update this in the future to be more correct.

### Testing Instructions
- Navigate to http://calypso.localhost:3000/comments/pending/
- Select a site that has no comments, or no comments that are pending
- Wait some time, we should see a notice that there are no comments
- Repeat steps above for a site with comments, we should see comments/paging behave normally
